scripts/BONSpecialGenerator.swift

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env swift
+
+//
+//  BONSpecialGenerator.swift
+//  BonMot
+//
+//  Created by Zev Eisenberg on 7/16/15.
+//
+
+import Foundation
+
+// Please keep this array sorted
+let specialCharacters: [unichar] = [
+    0x00A0,
+    0x2002,
+    0x2003,
+    0x2007,
+    0x2009,
+    0x200A,
+    0x200B,
+    0x2011,
+    0x2012,
+    0x2013,
+    0x2014,
+    0x202F,
+    0x2060,
+    0x2212,
+]
+
+extension unichar {
+    var unicodeName : String {
+        get {
+            let swiftCharacter = Character(UnicodeScalar(self))
+
+            let theCFMutableString = NSMutableString(string: String(swiftCharacter)) as CFMutableString
+            CFStringTransform(theCFMutableString, nil, kCFStringTransformToUnicodeName, 0)
+
+            let characterName = theCFMutableString as String
+            var trimmedName = characterName
+
+            if characterName != String(swiftCharacter) {
+                // characterName will look like "\N{NO-BREAK SPACE}", so trim "\N{" and "}"
+                trimmedName = characterName[3..<characterName.utf8.count - 1]
+            }
+
+            return trimmedName
+        }
+    }
+}
+
+extension String {
+    subscript(i: Int) -> Character {
+        return self[advance(startIndex, i)]
+    }
+
+    subscript(range: Range<Int>) -> String {
+        return self[advance(startIndex, range.startIndex)..<advance(startIndex, range.endIndex)]
+    }
+
+    func camelCaseMethodName() -> String {
+        let components: [String] = split(self.characters){$0 == " " || $0 == "-"}.map(String.init)
+        var camelCaseComponents = components.map { $0.capitalizedString }
+        if camelCaseComponents.count > 0 {
+            camelCaseComponents[0] = camelCaseComponents[0].lowercaseString
+        }
+        return "".join(camelCaseComponents)
+    }
+}
+
+// from http://stackoverflow.com/a/31480534/255489
+func pathToFolderContainingThisScript() -> String {
+    let cwd = NSFileManager.defaultManager().currentDirectoryPath
+
+    let script = Process.arguments[0];
+
+    if script.hasPrefix("/") { // absolute
+        let path = script.stringByDeletingLastPathComponent
+        return path
+    }
+    else { // relative
+        let urlCwd = NSURL(fileURLWithPath: cwd)
+
+        if let urlPath = NSURL(string: script, relativeToURL: urlCwd) {
+            if let path = urlPath.path {
+                let path = path.stringByDeletingLastPathComponent
+                return path
+            }
+        }
+    }
+
+    return ""
+}
+
+// ******************************
+// *                            *
+// *  Real program starts here  *
+// *                            *
+// ******************************
+
+// Make sure specialCharacters is sorted
+let sortedSpecialCharacters = specialCharacters.sort({$0 < $1})
+if sortedSpecialCharacters != specialCharacters {
```

Comment:
#? is this just to make sure no dupes are in the set? Couldn't you cast the initial array as a set, and sort that rather than worrying about if the input is sorted unless you're fielding a pull request?